biogeophys/EDSurfaceAlbedoMod.F90

```diff
@@ -195,756 +125,918 @@ subroutine ED_Norman_Radiation (nsites, sites, bc_in, bc_out )
+                           Dif_up(L,ft,iv) = Dif_up(L,ft,iv) + Dif_up(L,ft,iv+1) * &
+                                (ftweight(L,ft,1)-ftweight(L,ft,iv))/ftweight(L,ft,1)
+                           !nb is this the right constuction?
+                           ! the radiation that hits the empty space is not reflected.
+                        else
+                           Dif_up(L,ft,iv) = dif_ratio(L,ft,iv,ib) * Dif_dn(L,ft,iv) * ftweight(L,ft,iv)
+                           Dif_up(L,ft,iv) = Dif_up(L,ft,iv) + Dif_up(L,ft,iv+1) * (1.0_r8-ftweight(L,ft,iv))
+                        endif
+                     end do
+                     
+                     weighted_dif_up(L) = weighted_dif_up(L) + Dif_up(L,ft,1) * ftweight(L,ft,1)
+                     !instance where the first layer ftweight is used a proxy for the whole column. FTWA
+                  endif !present
+               end do !ft
+               if (L == currentPatch%NCL_p.and.currentPatch%NCL_p > 1)then !is this the (incomplete) understorey?
+                  !Add on the radiation coming up through the canopy gaps.
+                  !diffuse to diffuse
+                  weighted_dif_up(L) = weighted_dif_up(L) +(1.0-sum(ftweight(L,1:numpft,1))) * &
+                       weighted_dif_down(L-1) * currentPatch%gnd_alb_dif(ib) 
+                  !direct to diffuse
+                  weighted_dif_up(L) = weighted_dif_up(L) + forc_dir(radtype) * &
+                       weighted_dir_tr(L-1) * (1.0-sum(ftweight(L,1:numpft,1))) * currentPatch%gnd_alb_dir(ib)
+               endif
+            end do !L
+            
+            !++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++!
+            ! 3. Iterative calculation of forward and upward diffuse fluxes, iNCL_puding
+            ! scattered direct beam
+            !++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++!
+
+            ! Flag to exit iteration loop: 0 = exit and 1 = iterate
+            irep = 1
+            ! Iteration loop
+            iter = 0
+            do while(irep ==1 .and. iter<50)
+               
+               iter = iter + 1
+               irep = 0
+               do L = 1,currentPatch%NCL_p !working from the top down
+                  weighted_dif_down(L) = 0._r8
+                  do ft =1,numpft
+                     if (currentPatch%canopy_mask(L,ft) == 1)then
+                        ! forward diffuse flux within the canopy and at soil, working forward through canopy
+                        ! with Dif_up -from previous iteration-. Dif_dn(1) is the forward diffuse flux onto the canopy.
+                        ! Note: down = forward flux onto next layer
+                        if (L == 1)then !is this the top layer?
+                           Dif_dn(L,ft,1) = forc_dif(radtype)
+                        else
+                           Dif_dn(L,ft,1) = weighted_di
```

Comment:
```suggestion
                          (1.0_r8-sum(ftweight(L,1:numpft,1)))
```

Reply:
that suggestion got messed up somehow i think?  beta...

Reply:
think I got that in, done